### DISCUSSION TITLE
Automatically enable supporter leaderboard features after purchase

### DISCUSSION THREAD

**Discussion by DutchEllie** [ppy/osu#23354] [Ideas]
When you purchase osu! supporter from the website, you have to restart the game (if it is already open during purchase) to get the client to recognize your new status and enable the leaderboard features (like country and friend leaderboards). So this is how it happens now:
1. Have the game open and logged into your account
2. Buy supporter on the website on that account
3. Go back to the game, find that supporter features aren't enabled yet
4. Restart
5. Enjoy

I would like to suggest that this is changed to the following:
1. Have the game open and logged into your account
2. Buy supporter on the website on that account
3. Go back to the game, find all supporter features enabled immediately (and maybe a thank you message or something ingame upon purchase).

It's admittedly a very small detail, but for me that goes a long way! It's not something many people will experience, but when you do experience it, it's kinda annoying.

**Comment by bdach** [comment]
As things are this is going to be a little painful to make happen. We'd need to refactor how login/the supporter flag works, and additionally rope the spectator server into this so that the information about the supporter purchase is broadcast to the correct client (likely via redis pub-sub).

Would be nice to have, but unfortunately a substantial amount of work.

**Reply by peppy** [reply]
I think once we get notifications websockets hooked up, we'll receive event for things like "user has become supporter", which should make things quite easy.
